library/ssl_msg.c

```diff
@@ -3291,10 +3295,59 @@ int mbedtls_ssl_prepare_handshake_record(mbedtls_ssl_context *ssl)
         }
     } else
 #endif /* MBEDTLS_SSL_PROTO_DTLS */
-    /* With TLS we don't handle fragmentation (for now) */
-    if (ssl->in_msglen < ssl->in_hslen) {
-        MBEDTLS_SSL_DEBUG_MSG(1, ("TLS handshake fragmentation not supported"));
-        return MBEDTLS_ERR_SSL_FEATURE_UNAVAILABLE;
+    {
+        int ret;
+        const size_t hs_remain = ssl->in_hslen - ssl->in_hsfraglen;
+        const size_t msg_hslen = (hs_remain <= ssl->in_msglen ? hs_remain : ssl->in_msglen);
```

Comment:
I am not sure that this is ideal. This is a debug variable, used for logging, and could easily be a macro. This would also remove the need to void it later on.

Reply:
well, this was my attempt to pull out a part of a long expression for readability. but i did get yelled at by the compiler when build with debug disabled, hence the `(void)`. ok, i folded the expression into the logging statement (macro is ugly and breaks the flow of code).